src/coreclr/jit/assertionprop.cpp

```diff
@@ -4699,6 +4699,17 @@ GenTree* Compiler::optAssertionProp_Update(GenTree* newTree, GenTree* tree, Stat
             if (parent != nullptr)
             {
                 parent->ReplaceOperand(useEdge, newTree);
+                if ((parent->gtOper == GT_IND) && newTree->IsIconHandle())
+                {
+                    parent->gtFlags |= GTF_IND_NONFAULTING;
+                    GenTreeFlags handleKind = newTree->GetIconHandleFlag();
+
+                    if ((handleKind != GTF_ICON_STATIC_HDL) && (handleKind != GTF_ICON_BBC_PTR) &&
```

Comment:
this, gtNewIndir and debug-check have these condition - can you unify them all into a helper method or something?

Reply:
What would be the best place for the helper?

Reply:
Don't know, perhaps some GenTreeIndir::SetIndirFlags() ?